### ISSUE TITLE
Whether auth_getSession produces a session secret shouldn't depend on whether _isDesktop is set

### ISSUE THREAD

**Issue by GoogleCodeExporter** [kalyanhadooptraining/facebook-java-api#179]
```
As described in http://code.google.com/p/facebook-java-api/wiki/DesktopMode
, the server that is generating the secret session shouldn't be set in
desktop mode. It's only the desktop client itself that receives and uses
the secret session that should be marking itself as "Desktop mode". By
marking itself as Desktop, it will be notifying the API to expect failures
when calling some methods and allowing the API to handle these scenarios
smoothly. That's the only reason to bother to setIsDesktop(true).

Changes to be made:
  * Create a new version of auth_getSession(authToken, boolean
generateSecretSession) [note the additional boolean].
  * Default behaviour of normal auth_getSession(authToken) is just to call
with generateSecretSession = false.
  * Modify auth_getSession to not pay any attention to _isDesktop flag.

So, this requires a workflow change for users who are really trying to
generate secret session tokens. They'll have to change to using
auth_getSession(token, true) and not bother setting _isDesktop to true. A
such, the Javadoc needs to be improved considerably to explain this and it
needs to go on some release notes.

The server can use getCacheSessionSecret() to get hold of the session
secret key.

In summary, this means no changes for normal (web) API users and minor
changes for people trying to generate a desktop session secret. These
changes will clarify what the purpose of setIsDesktop() is and will reduce
confusion considerably. So, it's worth making the change...

```

Original issue reported on code.google.com by `david.j....@googlemail.com` on 12 Mar 2009 at 11:18
